### PR TITLE
[export][refactor][6/n] Remove equality_constraints

### PR DESCRIPTION
**Pull request by angelayi**
Through the new dynamic_shapes API and using torch.export.Dim, dimensions that are equal will now be represented by the same symbol, so we no longer need to store `equality_constraints`.

Differential Revision: D52351705

### CODE REVIEW

**Review by avikchaudhuri — APPROVED** [review]
How are you going to land this internally? Will this require a change to executorch as well?

**Comment by angelayi** [comment]
The internal diff [D52351705](https://www.internalfb.com/diff/D52351705) is on top of a stack of diffs that have already been landed which migrated ppl off of equality constraints. Since CI is green for both internal and external I expect this diff to be good to go? :>
